tern/analyze/common.py

```diff
@@ -452,3 +453,76 @@ def get_os_style(image_layer, binary):
                     package_manager=binary,
                     package_format=image_layer.pkg_format,
                     os_list=image_layer.os_guess), 'info'))
+
+
+def check_git_src(dockerfile_path, src_path):
+    '''Given the src_path and the dockerfile path, return the git
+    repository name and sha information in the format of string.'''
+    # get the path of the folder containing the dockerfile
+    dockerfile_folder_path = os.path.dirname(dockerfile_path)
+    # get the path of the target folder or file
+    src_path = os.path.join(dockerfile_folder_path, src_path)
```

Comment:
This `src_path` is incorrect here. You will want to use `get_git_toplevel(dockerfile_path)` for the `src_path` here and then you don't need to grab it later in line 472.

Reply:
I am considering this situation:
- dockerfile_folder
    - Dockerfile
    - git_project_folder

ADD command is like `ADD git_project_folder /tmp`. 

In this situation, args given to `check_git_src` equals: 
`src_path = git_project_folder`
`dockerfile_path = path_to_dockerfile_folder/dockerfile_folder/Dockerfile`.

So I use `os.path.join` to get the path `path_to_dockerfile_folder/dockerfile_folder/git_project_folder`.
If `src_path = get_git_toplevel(dockerfile_folder_path)`, we will get the toplevel directory containing this dockerfile instead of looking into the src given in ADD command first.

In another way, if ADD command is copying a folder which contains a git project(assume the name is A) into the image, and the dockerfile itself is in a git repo(assume the name is B), should we record the A or B?
- B
    - subfolder
        - Dockerfile
        - A(src of the ADD command)

Reply:
> I am considering this situation:
> 
>     * dockerfile_folder
>       
>       * Dockerfile
>       * git_project_folder
> 

Your code works for this situation, but this is a corner case. See below. 

> In another way, if ADD command is copying a folder which contains a git project(assume the name is A) into the image, and the dockerfile itself is in a git repo(assume the name is B), should we record the A or B?
> 
>     * B
>       
>       * subfolder
>         
>         * Dockerfile
>         * A(src of the ADD command)

We want to record the git information from the repo that is being copied, so we would want to record A. However, I think we should focus this PR for now on the following use case and we can add more later.

The most common use case we will see for this is:
* git_project_folder
    * directroy1
    * directory2/Dockerfile

ADD git_project_folder /tmp

which means that with your code now:
`dockerfile_folder_path = ~/git_project_folder/directory2`
`src_path = git_project_folder`
and thus, `src_path = ~/git_project_folder/directory2/git_project_folder`. If you check to see if this directory exists, it will return false and the comment will say "Unable to find the src file or filter".

Because the above is the most common case, I think we should just use `get_top_dir` of the dockerfile location and worry about the other corner cases later. We have a release next week and ideally, this functionality will be included. What do you think?

I also want to apologize, because I think I may have confused you a bit considering all of the situations. I am sorry for that and want you to know that you are doing an excellent job of contributing and making meaningful changes. Your efforts do not go unnoticed!

Reply:
Thinks for your explanation! Maybe I should figure out the use case clearly before coding. I am enjoying contributing and thanks for encouragement! I will work on it now!